src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'removeProperties':
+        action.keysToRemove.forEach(item => {
+          delete state[item];
+        });
+        break;
+      case 'clear':
+        Object.keys(state).forEach(key => {
+          delete state[key];
+        });
+        break;
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      default:
+        break;
+    }
+  }
+
+  return state;
```

Comment:
You don't need to return anything here 🙂

Reply:
agree